packages/collector/src/announceCycle/agentHostLookup.js

```diff
@@ -166,6 +164,51 @@ function checkHost(host, cb) {
   req.end();
 }
 
+/**
+ * Checks the response payload to determine whether we have actually connected to an Instana host agent.
+ *
+ * @param {string} host the target IP (will only be used for logging in this method)
+ * @param {import('http').IncomingMessage} res the HTTP response
+ * @param {(...args: *) => *} cb
+ */
+function handleResponse(host, res, cb) {
+  // We inspect the payload, checking whether it is JSON and has the "version" property, to verify that we have
+  // actually connected to an Instana host agent.
+
+  /** @type {Array.<Buffer>} */
+  const chunks = [];
+  res.on('data', chunk => {
+    chunks.push(chunk);
+  });
+  res.on('end', () => {
+    const responsePayload = String(Buffer.concat(chunks));
```

Comment:
Question: Which encoding is this? Wondering why we do not use `res.setEncoding('utf-8')`. Then we concat the received chunk strings and parse the JSON.

Reply:
It is utf-8. I wonder which is cheaper, collecting Buffer objects, concatenating them, and then converting to a string or converting each chunk and concatenating it. It probably does not matter at all. I'll go with your suggestion, that's also how the examples on https://nodejs.org/api/http.html are written. (If we found an agent it is only one chunk anyway, so it matters even less).

Reply:
Also wondering what is cheaper.